app/controllers/themes_controller.rb

```diff
@@ -1,2 +1,30 @@
 class ThemesController < ApplicationController
+  def index
+    @themes = Theme.all
+  end
+
+  def show
+    @theme = Theme.find(params[:id])
+  end
+
+  def new
+    @theme = Theme.new
+  end
+
+  def create
+    @theme = Theme.new(theme_params)
+    @theme.user = current_user
+    if @theme.save
+      redirect_to theme_path(@theme)
+    else
+      render 'theme/show'
```

Comment:
if we could not create, I think it will be better to render "new"

Reply:
Yes Vivien you are right if @theme.save is false then render :new so it displays the form again.